test/utils.test.js

```diff
@@ -1,60 +1,127 @@
-import { getPageValue } from '../src/utils';
-import { assert } from 'chai';
-
-describe('utils.js', () => {
-  describe('getPageValue', () => {
-    it('returns the highest in bounds page value when page is out of bounds and count is greater than rowsPerPage', () => {
-      const count = 30;
-      const rowsPerPage = 10;
-      const page = 5;
-
-      const actualResult = getPageValue(count, rowsPerPage, page);
-      assert.strictEqual(actualResult, 2);
-    });
-
-    it('returns the highest in bounds page value when page is in bounds and count is greater than rowsPerPage', () => {
-      const count = 30;
-      const rowsPerPage = 10;
-      const page = 1;
-
-      const actualResult = getPageValue(count, rowsPerPage, page);
-      assert.strictEqual(actualResult, 1);
-    });
-
-    it('returns the highest in bounds page value when page is out of bounds and count is less than rowsPerPage', () => {
-      const count = 3;
-      const rowsPerPage = 10;
-      const page = 1;
-
-      const actualResult = getPageValue(count, rowsPerPage, page);
-      assert.strictEqual(actualResult, 0);
-    });
-
-    it('returns the highest in bounds page value when page is in bounds and count is less than rowsPerPage', () => {
-      const count = 3;
-      const rowsPerPage = 10;
-      const page = 0;
-
-      const actualResult = getPageValue(count, rowsPerPage, page);
-      assert.strictEqual(actualResult, 0);
-    });
-
-    it('returns the highest in bounds page value when page is out of bounds and count is equal to rowsPerPage', () => {
-      const count = 10;
-      const rowsPerPage = 10;
-      const page = 1;
-
-      const actualResult = getPageValue(count, rowsPerPage, page);
-      assert.strictEqual(actualResult, 0);
-    });
-
-    it('returns the highest in bounds page value when page is in bounds and count is equal to rowsPerPage', () => {
-      const count = 10;
-      const rowsPerPage = 10;
-      const page = 0;
-
-      const actualResult = getPageValue(count, rowsPerPage, page);
-      assert.strictEqual(actualResult, 0);
-    });
-  });
-});
+import { assembleCSV, getPageValue } from '../src/utils';
+import { assert } from 'chai';
+
+describe('utils.js', () => {
+  describe('assembleCSV', () => {
+    const options = {
+      downloadOptions: {
+        separator: ';',
+      },
+      onDownload: null,
+    };
+
+    const columns = [
+      {
+        name: 'firstname',
+        download: true,
+      },
+      {
+        name: 'lastname',
+        download: true,
+      },
+    ];
+
+    describe('when given two rows', () => {
```

Comment:
So, I think we still have too many nested `describe`s. If you look at the `getPageValue` tests, you'll see that we just use one `describe` block for the whole function, breaking up the bits into `it` blocks instead. So, this one would become `it('renders a two-record csv when given two rows`... Let's do that for all the new tests here, just one `describe` for `assembleCSV` and everything else is an `it` block.